rotkehlchen/rotkehlchen.py

```diff
@@ -858,6 +850,26 @@ def set_settings(self, settings: ModifiableDBSettings) -> Tuple[bool, str]:
         if settings.current_price_oracles is not None:
             Inquirer().set_oracles_order(settings.current_price_oracles)
 
+        if settings.ethereum_nodes_to_connect is not None:
+            # the frontend only provides the weight and the name of the node to connect
+            # and we need to qery the database
+            nodes = settings.ethereum_nodes_to_connect
+            db_nodes = self.data.db.get_ethereum_nodes()
+            nodes_with_info = []
+            for node in nodes:
+                node_name = node.node_info.name
+                if node_name in db_nodes:
+                    new_node = WeightedNode(
+                        node_info=db_nodes[node_name],
+                        weight=node.weight,
+                    )
+                    nodes_with_info.append(new_node)
+                else:
+                    msg = f'Tried to set node {node_name} but information is not in the db'
```

Comment:
This entire logic in this PR is completely broken ... man what were you thinking?

You are putting some info in the settings .. some info in a new table .... the frontend gives you some data, you have to figure out the rest of the data ... ?

This is a mess ...

Please think and plan clearly when working on the rotki codebase. I need you to be proposing beautiful code that improves the architecture and wows me.

Not making complicated ugly code that I have to very carefully examine, determine what's wrong, figure out a better way to do it and tell you to rewrite it.

1. The nodes can get out of the settings. One place. Add the new table. But it should be more generic. Each node should have name, endpoint, chain id and weight. Figuring out if it's owned can also be done by an `in` check in python on a mapping of the hard coded nodes.
2. First time a user is created the default mapping is put into the the table. No need to populate it from the schema like you did.
3. The user can then get them, add more, edit, delete nodes.

Isn't that simpler?